bigframes/core/__init__.py

```diff
@@ -94,6 +95,7 @@ def __init__(
         hidden_ordering_columns: Optional[Sequence[ibis_types.Value]] = None,
         ordering: ExpressionOrdering = ExpressionOrdering(),
         predicates: Optional[Collection[ibis_types.BooleanValue]] = None,
+        api_methods: Dict[str, str] = {},
```

Comment:
I'm worried about adding this to ArrayValue. Making api method tracking something that cuts up and down the stack will make it tricky to maintain.

Reply:
For APIs that will call down the to the lower level, like aggregate here, we pass it down to the ArrayValue. For other cases, we don't need to do so. What potential maintenance problems are you thinking about here? We don't need to consider any acyclic graph, so the stacking is just for debugging purpose.

Reply:
Change it to list now.